src/components/ExpenseForm/ExpenseForm.jsx

```diff
@@ -0,0 +1,69 @@
+import React, { useState, useEffect } from 'react'
+import { DataService } from '../../services/DataService'
+import DatePicker from 'react-datepicker'
+import { expensesUrl } from '../../static/apiconfig'
+
+import "react-datepicker/dist/react-datepicker.css"
+
+import * as Styled from './style';
+
+const  ExpenseForm = () => {
+  const [cost, setCost] = useState('')
+  const [category, setCategory] = useState('')
+  const [date, setDate] = useState(new Date())
+  const [isSubmitting, setSubmitting] = useState(false)
+
+  useEffect(() => {
+    (async function fetchExpenses() {
+      if(!isSubmitting) { return }
+      try {
+        await DataService.postData(expensesUrl, { cost, category, date })
+        setSubmitting(false)
+      } catch (e) {
+        console.error(e)
+      }
+    })()
+  }, [isSubmitting]) // eslint-disable-line react-hooks/exhaustive-deps
+
+  const handleDateChange = date => {
```

Comment:
is ihis necessary? Can't you use setDate inside template?

Reply:
Changed